merqube_client_lib/api_client/base.py

```diff
@@ -132,26 +131,39 @@ def create_index(self, index_def: IndexDefinitionPost) -> dict[str, str]:
 
         TODO: examples and index templates to be added to this repo.
         """
-        payload = json.loads(index_def.json(exclude_none=True, exclude_defaults=True))
-        return cast(dict[str, str], self.session.post("/index", json=payload).json())
+        return cast(dict[str, str], self.session.post("/index", json=pydantic_to_dict(index_def)).json())
 
-    def update_index(self, index_id: str, index_def: Index) -> None:
+    def update_index(self, index_def: Index) -> dict[str, str]:
         """
         Update (full object replacement) an index
         """
-        self.session.put(f"/index/{index_id}", json=index_def.dict())
+        return cast(dict[str, str], self.session.put(f"/index/{index_def.id}", json=pydantic_to_dict(index_def)).json())
 
-    def patch_index(self, index_id: str, index_updates: Manifest) -> None:
+    def patch_index(self, index_id: str, index_updates: Manifest) -> dict[str, str]:
         """
         Patch an index - index_updates is a partial index manifest
         """
-        self.session.patch(f"/index/{index_id}", json=index_updates)
+        return cast(dict[str, str], self.session.patch(f"/index/{index_id}", json=index_updates).json())
 
-    def delete_index(self, index_id: str) -> None:
+    def delete_index(self, index_id: str) -> dict[str, str]:
         """
         Delete an index
         """
-        self.session.delete(f"/index/{index_id}")
+        return cast(dict[str, str], self.session.delete(f"/index/{index_id}").json())
+
+    def replace_target_portfolio(self, index_id: str, target_portfolio: dict[str, Any]) -> dict[str, Any]:
+        """
+        Target portfolios are a PUT, not a POST, because an index can only have one open portfolio on a given day.
+        "Last one wins".
+        It is not defined behavior to have two open portfolios on the same day.
```

Comment:
Does "open" here mean active? In the CITI case they wanted to be able to update the portfolio during the trading day, so it's not necessarily the open portfolio.

Maybe "...to have multiple portfolios effective at the same time"?

Reply:
good call, rewording